src/backend/distributed/executor/multi_utility.c

```diff
@@ -461,6 +462,81 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 							params, dest, completionTag);
 #endif
 
+#if (PG_VERSION_NUM >= 100000)
+	if (IsA(parsetree, CreateStmt))
+	{
+		CreateStmt *createStatement = (CreateStmt *) parsetree;
+
+		/*
+		 * If a partition is being created and if its parent is a distributed
+		 * table, we will distribute this table as well.
+		 */
+		if (createStatement->inhRelations != NIL && createStatement->partbound != NULL)
+		{
+			RangeVar *parentRelation = linitial(createStatement->inhRelations);
+			bool parentMissingOk = false;
+			Oid parentRelationId = RangeVarGetRelid(parentRelation, NoLock,
+													parentMissingOk);
+
+			if (IsDistributedTable(parentRelationId))
+			{
+				bool missingOk = false;
+				Oid relationId = RangeVarGetRelid(createStatement->relation, NoLock,
+												  missingOk);
+				Var *parentDistributionColumn = DistPartitionKey(parentRelationId);
+				char parentDistributionMethod = DISTRIBUTE_BY_HASH;
+				uint32 parentColocationId = TableColocationId(parentRelationId);
+
+				CreateDistributedTable(relationId, parentDistributionColumn,
+									   parentDistributionMethod, parentColocationId);
+			}
+		}
+	}
+	else if (IsA(parsetree, AlterTableStmt))
+	{
+		AlterTableStmt *alterTableStatement = (AlterTableStmt *) parsetree;
+		List *commandList = alterTableStatement->cmds;
+		ListCell *commandCell = NULL;
+
+		foreach(commandCell, commandList)
+		{
+			AlterTableCmd *alterTableCommand = (AlterTableCmd *) lfirst(commandCell);
+
+			/*
+			 * If a partition is being attached to a distributed table, we will distribute
+			 * attached table as well.
+			 */
+			if (alterTableCommand->subtype == AT_AttachPartition)
+			{
+				Oid relationId = AlterTableLookupRelation(alterTableStatement, NoLock);
```

Comment:
In here (and above) I preferred using relationId for the relation which is the subject of the query (and used parentRelationId or partitionRelationId for the other table (whichever is applicable).

For this case query is ALTER TABLE partitioned_table ATTACH PARTITION partition_table. Thus I used relationId for partitioned table. Since we have partitionRelationId in this code block. It might be confusing since both variable names are very similar.